### PR TITLE
Fixes 3420: add service notify for templates

### PR DESCRIPTION
**Pull request by rverdile**
## Summary
Sends an event on a new kafka topic when a template is created, updated, or deleted. 

This has nothing to do with sending an event (notification) to the notifications service or getting emails. I've used the term "notification" in places because it is still technically a notification (we're notifying the patch service), I wasn't sure what else to call it, and there's an opportunity here to share code between the templates and notifications. I'm open to other names!

## Testing steps
1. In a terminal, `make kafka-topic-consume KAFKA_TOPIC=platform.content-sources.template`
2. Start the API server
3. Create a template. Check the terminal where you've just run the kafka consumer, you should see a notification.
4. Update a template. Check for a notification.
5. Delete a template. Check for a notification.
## Checklist

- [ ] Tested with snapshotting feature disabled and pulp server URL not configured if appropriate

### CODE REVIEW

File: pkg/notifications/client.go [hunk 1]
```diff
@@ -77,3 +77,34 @@ func SetEmptyToNil(value string) *string {
...
+	if config.Get().TemplatesNotificationsClient != nil && len(templates) > 0 {
+		eventNameStr := eventName.String()
+		newUUID, _ := uuid.NewRandom()
+		e := cloudevents.NewEvent()
+		e.SetSource("urn:redhat:source:console:app:repositories")
+		e.SetID(newUUID.String())
+		e.SetType("com.redhat.console.repositories." + eventNameStr)
+		e.SetSubject("urn:redhat:subject:console:rhel:" + eventNameStr)
+		e.SetTime(time.Now())
+		e.SetExtension("redhatorgid", orgID)
```
Comment (rverdile) [inline pkg/notifications/client.go:92]: These fields are set for the repository notifications. Not sure it's super important here, but I've copied them over as they seem like they might contain useful information.

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-3420

File: pkg/config/config.go [hunk 2]
```diff
@@ -219,6 +221,7 @@ func setDefaults(v *viper.Viper) {
 	v.SetDefault("options.introspect_api_time_limit_sec", DefaultIntrospectApiTimeLimitSec)
 	v.SetDefault("options.always_run_cron_tasks", false)
 	v.SetDefault("options.enable_notifications", false)
+	v.SetDefault("options.enable_templates_notifications", false)
```
Comment (jlsherrill) [inline pkg/config/config.go:224]: we should probably default this to true.  We set notifications to false as we didn't want it turned on in production.   In fact, i might would argue we make the topic configurable (defaulting to 'platform.content-sources.template') and send it if its set.    That way, you're implicitly turning it on by setting the topic to deliver it on.

File: pkg/config/event.go [hunk 3]
```diff
@@ -74,30 +75,81 @@ func readEnv(key string, def string) string {
...
+
+	c, err := v2.NewClient(protocol, v2.WithTimeNow(), v2.WithUUIDs())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cloud events client: %w", err)
+	}
+	return c, nil
+}
+
+// SetupTemplatesNotifications creates the cloud events kafka client for sending events to the patch service
+func SetupTemplatesNotifications() {
```
Comment (jlsherrill) [inline pkg/config/event.go:106]: I think i'd perfer if we referred to these as events or messages internally.  Notifications is a console dot thing that is implemented via kafka events/messaging, but also had a rest implementation as well.

Comment (jlsherrill) [reply pkg/config/event.go:106]: (and to be clear, i think its fine to call kafka messages sent to platform.notifications.ingress as notifications, but calling it all generically 'notifications' might be confusing)

File: pkg/config/event.go [hunk 4]
```diff
@@ -74,30 +75,81 @@ func readEnv(key string, def string) string {
...
+
+func GetSaramaConfig() (*sarama.Config, error) {
 	saramaConfig := sarama.NewConfig()
 
 	saramaConfig.Version = sarama.V2_0_0_0
 	saramaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
 
-	if strings.Contains(LoadedConfig.Kafka.Sasl.Protocol, "SSL") {
-		saramaConfig.Net.TLS.Enable = true
-	}
```
Comment (jlsherrill) [inline pkg/config/event.go:94]: was this unnecessary?

Comment (rverdile) [reply pkg/config/event.go]: good catch

Comment (rverdile) [reply pkg/config/event.go]: I think I just missed it copying/pasting

File: pkg/config/event.go (hunk 3 above)
Comment (rverdile) [reply pkg/config/event.go]: agreed, it could be confusing. I renamed everything to "events".

Comment (rverdile) [reply pkg/config/event.go]: I also renamed `pkg/event -> pkg/kafka` and `pkg/notifications -> pkg/event`. I think these names better represent what the packages were being used for.

**Comment by jlsherrill** [comment]
You'll need to add the new topic name here:  https://github.com/content-services/content-sources-backend/blob/main/deployments/deployment.yaml#L21-L27

Otherwise this looks good and behave as expected !

**Review by jlsherrill — COMMENTED** [review]
ACK, but we may need to reach out to Qe about the test failure.  I'll rekick it for good measure

**Review by jlsherrill — APPROVED** [review]

**Comment by swadeley** [comment]
> ACK, but we may need to reach out to Qe about the test failure. I'll rekick it for good measure

Hi, the one fail in the test run is related to the read-only user in the rbac test. There have been some changes to user creation code. So, not related to this PR.

**Pull request merged**